src/lib/log/log.c

```diff
@@ -834,6 +834,23 @@ logs_free_all(void)
    * that happened between here and the end of execution. */
 }
 
+/** Close signal-safe log files. */
+void
+logs_close_minimal(void)
+{
+  logfile_t *victim, *next;
+  /* We can't lock the logs in a signal handler, or deallocate memory */
+  next = logfiles;
+  logfiles = NULL;
+  while (next) {
+    victim = next;
+    next = next->next;
+    if (victim->needs_close) {
+      close_log(victim);
```

Comment:
It might be safer to split close_log() into close_log() and close_log_sigsafe().

Reply:
Fixed in 671ec51163ec69e8b632bb121ae60354b9bd5b1d